From 4706e6fe7308556e2eb1ea6d8adf8047dcec82b9 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Thu, 18 Jul 2024 19:18:43 +0200
Subject: [PATCH] Enable Dogecoin addresses and private keys (#32)

Currently we still display BTC addresses, however, Dogecoin has their own prefixes (30 for P2PKH and 22 for P2SH).

Also update the private keys, which have the 158 prefix.

diff --git a/src/init.cpp b/src/init.cpp
index 54a4598a541a761ee039f396e0e68533469f72bf..0b2c28eb82a3d2338db7ce875e1a2b1f5c84f4a8 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -143,7 +143,7 @@ static const bool DEFAULT_PROXYRANDOMIZE = true;
 static const bool DEFAULT_REST_ENABLE = false;
 static constexpr bool DEFAULT_STOPAFTERBLOCKIMPORT{false};
 static constexpr bool DEFAULT_CHRONIK = false;
-static constexpr bool DEFAULT_USEASHADDR = true;
+static constexpr bool DEFAULT_USEASHADDR = false;
 
 #ifdef WIN32
 // Win32 LevelDB doesn't use filedescriptors, and the ones used for accessing
@@ -2899,7 +2899,7 @@ bool AppInitMain(Config &config, RPCServer &rpcServer,
     validation_signals.RegisterValidationInterface(node.peerman.get());
 
     // Encoded addresses using cashaddr instead of base58.
-    // We do this by default to avoid confusion with BTC addresses.
+    // Dogecoin uses distinct base58 prefixes, avoiding Bitcoin ambiguity.
     config.SetCashAddrEncoding(
         args.GetBoolArg("-usecashaddr", DEFAULT_USEASHADDR));
 
diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index 01efc121f50e00015a6a7c6f5c7ec7c6c02ab3e5..6bd429e9c44f8d7aae7777c9f45ab34753e3f4f3 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -179,9 +179,9 @@ public:
         vSeeds.emplace_back("seed.multidoge.org");
         vSeeds.emplace_back("seed2.multidoge.org");
 
-        base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 0);
-        base58Prefixes[SCRIPT_ADDRESS] = std::vector<uint8_t>(1, 5);
-        base58Prefixes[SECRET_KEY] = std::vector<uint8_t>(1, 128);
+        base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 30);
+        base58Prefixes[SCRIPT_ADDRESS] = std::vector<uint8_t>(1, 22);
+        base58Prefixes[SECRET_KEY] = std::vector<uint8_t>(1, 158);
         base58Prefixes[EXT_PUBLIC_KEY] = {0x04, 0x88, 0xB2, 0x1E};
         base58Prefixes[EXT_SECRET_KEY] = {0x04, 0x88, 0xAD, 0xE4};
         cashaddrPrefix = opts.ecash ? "ecash" : "bitcoincash";
@@ -330,9 +330,9 @@ public:
         // nodes with support for servicebits filtering should be at the top
         vSeeds.emplace_back("testseed.jrn.me.uk");
 
-        base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 111);
+        base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 113);
         base58Prefixes[SCRIPT_ADDRESS] = std::vector<uint8_t>(1, 196);
-        base58Prefixes[SECRET_KEY] = std::vector<uint8_t>(1, 239);
+        base58Prefixes[SECRET_KEY] = std::vector<uint8_t>(1, 241);
         base58Prefixes[EXT_PUBLIC_KEY] = {0x04, 0x35, 0x87, 0xCF};
         base58Prefixes[EXT_SECRET_KEY] = {0x04, 0x35, 0x83, 0x94};
         cashaddrPrefix = opts.ecash ? "ectest" : "bchtest";
diff --git a/src/qt/test/guiutiltests.cpp b/src/qt/test/guiutiltests.cpp
index b4abce9ded26807f7b26e1f8f70fe3406ec635e5..23ce34929e60a2bff4153185505c257786d69abb 100644
--- a/src/qt/test/guiutiltests.cpp
+++ b/src/qt/test/guiutiltests.cpp
@@ -47,7 +47,7 @@ void GUIUtilTests::toCurrentEncodingTest() {
     QString cashaddr_pubkey =
         "ecash:qpm2qsznhks23z7629mms6s4cwef74vcwva87rkuu2";
 
-    QString base58_pubkey = "1BpEi6DfDAUFd7GtittLSdBeYJvcoaVggu";
+    QString base58_pubkey = "DFxLFMAJWaNYA7TVTUstzPMFRSevAwTSLq";
 
     QVERIFY(GUIUtil::convertToCashAddr(params, cashaddr_pubkey) ==
             cashaddr_pubkey);
diff --git a/src/qt/test/uritests.cpp b/src/qt/test/uritests.cpp
index b30f718268a81647818ef8bf5ff5d67f47cf02b9..4bd9c64ea29536275ce24482421c5645d5961539 100644
--- a/src/qt/test/uritests.cpp
+++ b/src/qt/test/uritests.cpp
@@ -123,7 +123,7 @@ void URITests::uriTestFormatURI() {
     {
         // Legacy addresses are converted.
         SendCoinsRecipient r;
-        r.address = "12c6DSiU4Rq3P4ZxziKxzrL5LmMBrzjrJX";
+        r.address = "D6kBkhf7MqjKv4kZjJKXYcVgDu5VAyukJz";
         r.message = "test";
         QString uri = GUIUtil::formatBitcoinURI(*params, r);
         QVERIFY(uri == "ecash:qqgekzvw96vq5g57zwdfa5q6g609rrn0yccu9hrtvr?"
diff --git a/src/test/bloom_tests.cpp b/src/test/bloom_tests.cpp
index 805fbc9ce13001af53acb4394d8aea4da72f1af2..f22169dd7bc771bc138c459f24350eaeefad1241 100644
--- a/src/test/bloom_tests.cpp
+++ b/src/test/bloom_tests.cpp
@@ -107,7 +107,7 @@ BOOST_AUTO_TEST_CASE(bloom_create_insert_serialize_with_tweak) {
 
 BOOST_AUTO_TEST_CASE(bloom_create_insert_key) {
     std::string strSecret =
-        std::string("5Kg1gnAjaLfKiwhhPpGS3QfRg2m6awQvaj98JCZBZQ5SuS2F15C");
+        std::string("6KzLyjdh96x7mZ8rbCtQf2ef9TEe64hCBuGMchC5P4H5i4eTmD4");
     CKey key = DecodeSecret(strSecret);
     CPubKey pubkey = key.GetPubKey();
     std::vector<uint8_t> vchPubKey(pubkey.begin(), pubkey.end());
diff --git a/src/test/data/key_io_valid.json b/src/test/data/key_io_valid.json
index 175b297a1cac0eb62aa4dc7ee8fad121bfb0d2f7..9350b61919c9eabc57ff27fdc2e373ba7494196b 100644
--- a/src/test/data/key_io_valid.json
+++ b/src/test/data/key_io_valid.json
@@ -1,6 +1,6 @@
 [
     [
-        "1AGNa15ZQXAZUgFiqJ2i7Z2DPU2J6hW62i",
+        "DEQU7G2Chw4r1gSKZt2GfKBpGbkbQSwp3S",
         "76a91465a16059864a2fdbc7c99a4723a8395bc6f188eb88ac",
         {
             "isPrivkey": false,
@@ -8,7 +8,7 @@
         }
     ],
     [
-        "3CMNFxN1oHBc4R1EpboAL5yzHGgE611Xou",
+        "A36czoRusM4VxnNiEjTaaDcMyr4GB73xw1",
         "a91474f209f6ea907e2ea48f74fae05782ae8a66525787",
         {
             "isPrivkey": false,
@@ -16,7 +16,7 @@
         }
     ],
     [
-        "mo9ncXisMeAoXwqcV5EWuyncbmCcQN4rVs",
+        "nbpzakKSn16ZAp7nXuu9tELBrmiVpEyV1Q",
         "76a91453c0307d6851aa0ce7825ba883c6bd9ad242b48688ac",
         {
             "isPrivkey": false,
@@ -40,7 +40,7 @@
         }
     ],
     [
-        "5Kd3NBUAdUnhyzenEwVLy9pBKxSwXvE9FMPyR4UKZvpe6E3AgLr",
+        "6KwNf8w8CF5W2c5wSL7KamoQoNvV33WQrXXCjZ7DPb2GtvYj7X6",
         "eddbdc1168f1daeadbd3e44c1e3f8f5a284c2029f78ad26af98583a499de5b19",
         {
             "isCompressed": false,
@@ -49,7 +49,7 @@
         }
     ],
     [
-        "Kz6UJmQACJmLtaQj5A3JAge4kVTNQ8gbvXuwbmCj7bsaabudb3RD",
+        "QRVPTcD9SvFU86omfRt63uUgDXUwSgpQonbrPPbZqxUw2Xq9s1BY",
         "55c9bccb9ed68446d1b75273bbce89d7fe013a8acd1625514420fb2aca1a21c4",
         {
             "isCompressed": true,
@@ -58,7 +58,7 @@
         }
     ],
     [
-        "9213qJab2HNEpMpYNBa7wHGFKKbkDn24jpANDs2huN3yi4J11ko",
+        "95tYBmUej8ih5zyxBLp7qvTo9YqBXBhpK6JaiAQVtg4khJDdTEr",
         "36cb93b9ab1bdabf7fb9f2c04f1b9cc879933530ae7842398eef5a63a56800c2",
         {
             "isCompressed": false,
@@ -76,7 +76,7 @@
         }
     ],
     [
-        "cTpB4YiyKiBcPxnefsDpbnDxFDffjqJob8wGCEDXxgQ7zQoMXJdH",
+        "ckySCsSmg21tLzpBn21cC8Qvk64yQfrFv2BXK8uiwduYZ58KGxnP",
         "b9f4892c9e8282028fea1d2667c4dc5213564d41fc5783896a0d843fc15089f3",
         {
             "isCompressed": true,
@@ -94,7 +94,7 @@
         }
     ],
     [
-        "1Ax4gZtb7gAit2TivwejZHYtNNLT18PUXJ",
+        "DF6ADpqER651R2eKfXeJ73iVFW4kRN89jY",
         "76a9146d23156cbbdcc82a5a47eee4c2c7c583c18b6bf488ac",
         {
             "isPrivkey": false,
@@ -102,7 +102,7 @@
         }
     ],
     [
-        "3QjYXhTkvuj8qPaXHTTWb5wjXhdsLAAWVy",
+        "AFUoGYXezyc2jkwzhb7vqDa7EH1uQsohmE",
         "a914fcc5460dd6e2487c7d75b1963625da0e8f4c597587",
         {
             "isPrivkey": false,
@@ -110,7 +110,7 @@
         }
     ],
     [
-        "n3ZddxzLvAY9o7184TB4c6FJasAybsw4HZ",
+        "nrEqcBavLXTuRyHJ7HqhaLnsqsgs2gDdzz",
         "76a914f1d470f9b02370fdec2e6b708b08ac431bf7a5f788ac",
         {
             "isPrivkey": false,
@@ -126,7 +126,7 @@
         }
     ],
     [
-        "5K494XZwps2bGyeL71pWid4noiSNA2cfCibrvRWqcHSptoFn7rc",
+        "6KNUMV2uPdKPKb5VJQSVLF42H8uuf9tvotj6Ev9jRweThY81pje",
         "a326b95ebae30164217d7a7f57d72ab2b54e3be64928a19da0210b9568d4015e",
         {
             "isCompressed": false,
@@ -135,7 +135,7 @@
         }
     ],
     [
-        "L1RrrnXkcKut5DEMwtDthjwRcTTwED36thyL1DebVrKuwvohjMNi",
+        "QSpn1dLjrwQ1JjdQYA4gaxn35VVWGmAumxfEnr3SECwGPrjvwXTN",
         "7d998b45c219a1e38e99e7cbd312ef67f77a455a9b50c730c27f02c6f730dfb4",
         {
             "isCompressed": true,
@@ -144,7 +144,7 @@
         }
     ],
     [
-        "93DVKyFYwSN6wEo3E2fCrFPUp17FtrtNi2Lf7n4G3garFb16CRj",
+        "976ygS9ceHiZCsxT3BuCktb2eELhCGa8HJUsc5S42zbdEjwVrsp",
         "d6bca256b5abc5602ec2e1c121a08b0da2556587430bcf7e1898af2224885203",
         {
             "isCompressed": false,
@@ -153,7 +153,7 @@
         }
     ],
     [
-        "cTDVKtMGVYWTHCb1AFjmVbEbWjvKpKqKgMaR3QJxToMSQAhmCeTN",
+        "ckNkUD54qrLjEEcYGQXZ5wRa1cKdVANn1EpgAK19Skrrxq5FtkLx",
         "a81ca4e8f90181ec4b61b6a7eb998af17b2cb04de8a03b504b9e34c4c61db7d9",
         {
             "isCompressed": true,
@@ -162,7 +162,7 @@
         }
     ],
     [
-        "1C5bSj1iEGUgSTbziymG7Cn18ENQuT36vv",
+        "DGDgyyxMXgNxyTnbTZkpexwc1N6iH2RpPf",
         "76a9147987ccaa53d02c8873487ef919677cd3db7a691288ac",
         {
             "isPrivkey": false,
@@ -170,7 +170,7 @@
         }
     ],
     [
-        "3AnNxabYGoTxYiTEZwFEnerUoeFXK2Zoks",
+        "A1XdhRfSLsLrT5phz4uf2nUrWDdZRbGV7v",
         "a91463bcc565f9e68ee0189dd5cc67f1b0e5f02f45cb87",
         {
             "isPrivkey": false,
@@ -178,7 +178,7 @@
         }
     ],
     [
-        "n3LnJXCqbPjghuVs8ph9CYsAe4Sh4j97wk",
+        "nr1zGjoR1kfSLmn3BfMnAoQju4xaWE9VdH",
         "76a914ef66444b5b17f14e8fae6e7e19b045a78c54fd7988ac",
         {
             "isPrivkey": false,
@@ -194,7 +194,7 @@
         }
     ],
     [
-        "5KaBW9vNtWNhc3ZEDyNCiXLPdVPHCikRxSBWwV9NrpLLa4LsXi9",
+        "6KtWo7PLTGfVeezPRMzBL9Kd6urphr2hZcJkFynGgUXyNiebY3A",
         "e75d936d56377f432f404aabb406601f892fd49da90eb6ac558a733c93b47252",
         {
             "isCompressed": false,
@@ -203,7 +203,7 @@
         }
     ],
     [
-        "L1axzbSyynNYA8mCAhzxkipKkfHtAXYF4YQnhSKcLV8YXA874fgT",
+        "QSyt9SFyEPrfPfAEkyqkdwewDhKTD5g3wo6hV4iT4qjty66wuDi9",
         "8248bd0375f2f75d7e274ae544fb920f51784480866b102384190b1addfbaa5c",
         {
             "isCompressed": true,
@@ -212,7 +212,7 @@
         }
     ],
     [
-        "927CnUkUbasYtDwYwVn2j8GdTuACNnKkjZ1rpZd2yBB1CLcnXpo",
+        "95zh8weYJSE19s6xkf22dmUBJ8PdgC1WJqA5JrzpxVBnBX3ZrM1",
         "44c4f6a096eac5238291a94cc24c01e3b19b8d8cef72874a079e00a242237a52",
         {
             "isCompressed": false,
@@ -221,7 +221,7 @@
         }
     ],
     [
-        "cUcfCMRjiQf85YMzzQEk9d1s5A4K7xL5SmBCLrezqXFuTVefyhY7",
+        "cmmvLg9Y4iVQ2aPY6Z2XjyCqa2TcnnsXmeRTTmMBpUmL29xSZtKp",
         "d1de707020a9059d6d3abaf85e17967c6555151143db13dbb06db78df0f15c69",
         {
             "isCompressed": true,
@@ -230,7 +230,7 @@
         }
     ],
     [
-        "1Gqk4Tv79P91Cc1STQtU3s1W6277M2CVWu",
+        "DLyqbirkSo3HjcC3Bzt2bdB6y9qQjNKyxN",
         "76a914adc1cc2081a27206fae25792f28bbc55b831549d88ac",
         {
             "isPrivkey": false,
@@ -238,7 +238,7 @@
         }
     ],
     [
-        "33vt8ViH5jsr115AGkW6cEmEz9MpvJSwDk",
+        "9tg8sLnB9okjuNSdgtAWrNPcgijs1yMwBE",
         "a914188f91a931947eddd7432d6e614387e32b24470987",
         {
             "isPrivkey": false,
@@ -246,7 +246,7 @@
         }
     ],
     [
-        "mhaMcBxNh5cqXm4aTQ6EcVbKtfL6LGyK2H",
+        "nWFZaQYx7SYbAdLkWEksak8u9fqyrZLGvp",
         "76a9141694f5bc1a7295b600f40018a618a6ea48eeb49888ac",
         {
             "isPrivkey": false,
@@ -262,7 +262,7 @@
         }
     ],
     [
-        "5HtH6GdcwCJA4ggWEL1B3jzBBUB8HPiBi9SBc5h9i4Wk4PSeApR",
+        "6JCcPE6aVxax7J7fRid9fMyQetefnWzTKKZQvaL3XiiNs2kRoxP",
         "091035445ef105fa1bb125eccfb1882f3fe69592265956ade751fd095033d8d0",
         {
             "isCompressed": false,
@@ -271,7 +271,7 @@
         }
     ],
     [
-        "L2xSYmMeVo3Zek3ZTsv9xUrXVAmrWxJ8Ua4cw8pkfbQhcEFhkXT8",
+        "QUMMhcAdkQXgtGSc49kwqhh8xCoRZWRwMpkXimDbPx244AG42J8w",
         "ab2b4bcdfc91d34dee0ae2a8c6b6668dadaeb3a88b9859743156f462325187af",
         {
             "isCompressed": true,
@@ -280,7 +280,7 @@
         }
     ],
     [
-        "92xFEve1Z9N8Z641KQQS7ByCSb8kGjsDzw6fAmjHN1LZGKQXyMq",
+        "96qjbPY5FziapjDR8ZeS1qAkGpNBa9YyaDEsf575MKMLFZfHKhn",
         "b4204389cef18bbe2b353623cbf93e8678fbc92a475b664ae98ed594e6cf0856",
         {
             "isCompressed": false,
@@ -298,7 +298,7 @@
         }
     ],
     [
-        "cVM65tdYu1YK37tNoAyGoJTR13VBYFva1vg9FLuPAsJijGvG6NEA",
+        "cnWMEDMMFKNaz9uuuKm4PeePVutVD6U2LovQNFba9pp9HwPGwrxU",
         "e7b230133f1b5489843260236b06edca25f66adb1be455fbd38d4010d48faeef",
         {
             "isCompressed": true,
@@ -307,7 +307,7 @@
         }
     ],
     [
-        "1JwMWBVLtiqtscbaRHai4pqHokhFCbtoB4",
+        "DP5T3SRzC8kBQcnB9saGcaztgtRYd287bv",
         "76a914c4c1b72491ede1eedaca00618407ee0b772cad0d88ac",
         {
             "isPrivkey": false,
@@ -315,7 +315,7 @@
         }
     ],
     [
-        "3QCzvfL4ZRvmJFiWWBVwxfdaNBT8EtxB5y",
+        "AExFfWPxdVofCd5yvKANCoFx4kqAKxY4ap",
         "a914f6fe69bcb548a829cce4c57bf6fff8af3a5981f987",
         {
             "isPrivkey": false,
@@ -323,7 +323,7 @@
         }
     ],
     [
-        "mizXiucXRCsEriQCHUkCqef9ph9qtPbZZ6",
+        "nXfjh8D6qZnzVagNLKQqouCj5hfjH4ssH9",
         "76a914261f83568a098a8638844bd7aeca039d5f2352c088ac",
         {
             "isPrivkey": false,
@@ -339,7 +339,7 @@
         }
     ],
     [
-        "5KQmDryMNDcisTzRp3zEq9e4awRmJrEVU1j5vFRTKpRNYPqYrMg",
+        "6Kj6WpSJvyuWv5Rb1ScDSmdJ4MuJoyWm5BrKEk4M9Ud1M35KXvE",
         "d1fab7ab7385ad26872237f1eb9789aa25cc986bacc695e07ac571d6cdac8bc0",
         {
             "isCompressed": false,
@@ -348,7 +348,7 @@
         }
     ],
     [
-        "L39Fy7AC2Hhj95gh3Yb2AU5YHh1mQSAHgpNixvm27poizcJyLtUi",
+        "QUYB7wyBGuBrNc5jdpRp3gv9kj3LSzJ6a54dkZ9rrBR5SYEbzm2K",
         "b0bbede33ef254e8376aceb1510253fc3550efd0fcf84dcd0c9998b288f166b3",
         {
             "isCompressed": true,
@@ -357,7 +357,7 @@
         }
     ],
     [
-        "91cTVUcgydqyZLgaANpf1fvL55FH53QMm4BsnCADVNYuWuqdVys",
+        "95VwqwWkgVCRpyqyyY4evK7suJUiNT67LLL6GVY1UgZgWDzMNrZ",
         "037f4192c630f399d9271e26c575269b1d15be553ea1a7217f0cb8513cef41cb",
         {
             "isCompressed": false,
@@ -366,7 +366,7 @@
         }
     ],
     [
-        "cQspfSzsgLeiJGB2u8vrAiWpCU4MxUT6JseWo2SjXy4Qbzn2fwDw",
+        "ci35omig2eUzFJCa1Hidm4hnhLTfdJzYdktmuw8vWvZqAfAxMZER",
         "6251e205e8ad508bab5596bee086ef16cd4b239e0cc0c5d7c4e6035441e7d5de",
         {
             "isCompressed": true,
@@ -375,7 +375,7 @@
         }
     ],
     [
-        "19dcawoKcZdQz365WpXWMhX6QCUpR9SY4r",
+        "DDmi8CjxuyXhX3GgFQX4uTghHLD7fgjy12",
         "76a9145eadaf9bb7121f0f192561a5a62f5e5f5421029288ac",
         {
             "isPrivkey": false,
@@ -383,7 +383,7 @@
         }
     ],
     [
-        "37Sp6Rv3y4kVd1nQ1JV5pfqXccHNyZm1x3",
+        "9xC4qGyx38dPXP9sRS9W4oTuKBfR5ueTuH",
         "a9143f210e7277c899c3a155cc1c90f4106cbddeec6e87",
         {
             "isPrivkey": false,
@@ -391,7 +391,7 @@
         }
     ],
     [
-        "myoqcgYiehufrsnnkqdqbp69dddVDMopJu",
+        "nnV3au9J54qRVk4xogJUa4dite9Ngd1Est",
         "76a914c8a3c2a09a298592c3e180f02487cd91ba3400b588ac",
         {
             "isPrivkey": false,
@@ -407,7 +407,7 @@
         }
     ],
     [
-        "5KL6zEaMtPRXZKo1bbMq7JDjjo1bJuQcsgL33je3oY8uSJCR5b4",
+        "6KeSHC3KT9iKbwEAnyyoivCyDDV8p2gtUrTGNEGwdCLYF18xSu2",
         "c7666842503db6dc6ea061f092cfb9c388448629a6fe868d068c42a488b478ae",
         {
             "isCompressed": false,
@@ -416,7 +416,7 @@
         }
     ],
     [
-        "KwV9KAfwbwt51veZWNscRTeZs9CKpojyu1MsPnaKTF5kz69H1UN2",
+        "QNt4U1UvrZNCFT3c6eiQJgVBLBDtsMsnnG3nBQyABbh7S29TGFZP",
         "07f0803fc5399e773555ab1e8939907e9badacc17ca129e67a2f5f2ff84351dd",
         {
             "isCompressed": true,
@@ -425,7 +425,7 @@
         }
     ],
     [
-        "93N87D6uxSBzwXvpokpzg8FFmfQPmvX4xHoWQe3pLdYpbiwT5YV",
+        "97FcTfzyfHYTDB6Ecv4zamSobtdq5LCpXZwitwRcKwZbauT7fmD",
         "ea577acfb5d1d14d3b7b195c321566f12f87d2b77ea3a53f68df7ebf8604a801",
         {
             "isCompressed": false,
@@ -434,7 +434,7 @@
         }
     ],
     [
-        "cMxXusSihaX58wpJ3tNuuUcZEQGt6DKJ1wEpxys88FFaQCYjku9h",
+        "cf7o4CAX3tMM5yqqA3AhVpoXjGgBm3rkLpV65tZK7Ckzxrwc2SPh",
         "0b3b34f0958d8a268193a9814da92c3e8b58b4a4378a542863e34ac289cd830c",
         {
             "isCompressed": true,
@@ -443,7 +443,7 @@
         }
     ],
     [
-        "13p1ijLwsnrcuyqcTvJXkq2ASdXqcnEBLE",
+        "D7x7FzHbBCkuSz2DCWJ6JbBmKmG91E3nRZ",
         "76a9141ed467017f043e91ed4c44b4e8dd674db211c4e688ac",
         {
             "isPrivkey": false,
@@ -451,7 +451,7 @@
         }
     ],
     [
-        "3ALJH9Y951VCGcVZYAdpA3KchoP9McEj1G",
+        "A15Z1zc395N6Ays2xJJEQAwzQNmBNdDHY7",
         "a9145ece0cadddc415b1980f001785947120acdb36fc87",
         {
             "isPrivkey": false,
diff --git a/src/test/descriptor_tests.cpp b/src/test/descriptor_tests.cpp
index 745f4532d03d426b62ce93200163099966c23be2..b29c63fbb1a6c3cc4d0466c9aa36513ad0dea61d 100644
--- a/src/test/descriptor_tests.cpp
+++ b/src/test/descriptor_tests.cpp
@@ -361,7 +361,7 @@ BOOST_FIXTURE_TEST_SUITE(descriptor_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(descriptor_test) {
     // Basic single-key compressed
-    Check("combo(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+    Check("combo(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
           "combo("
           "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
           SIGNABLE,
@@ -369,7 +369,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "bdac",
             "76a9149a1c78a507689f6f54b847ad1cef1e614ee23f1e88ac"}},
           std::nullopt);
-    Check("pk(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+    Check("pk(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
           "pk("
           "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
           SIGNABLE,
@@ -377,7 +377,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "bdac"}},
           std::nullopt);
     Check("pkh([deadbeef/1/2'/3/4']"
-          "L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+          "QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
           "pkh([deadbeef/1/2'/3/4']"
           "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
           SIGNABLE, {{"76a9149a1c78a507689f6f54b847ad1cef1e614ee23f1e88ac"}},
@@ -385,7 +385,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
     // Missing start bracket in key origin
     CheckUnparsable(
         "pkh(deadbeef/1/2'/3/"
-        "4']L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+        "4']QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
         "pkh(deadbeef/1/2'/3/"
         "4']"
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
@@ -394,7 +394,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
     // Multiple end brackets in key origin
     CheckUnparsable(
         "pkh([deadbeef]/1/2'/3/"
-        "4']L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+        "4']QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
         "pkh([deadbeef]/1/2'/3/"
         "4']"
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
@@ -402,7 +402,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
 
     // Basic single-key uncompressed
     Check(
-        "combo(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+        "combo(6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
         "combo("
         "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8d"
         "ec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)",
@@ -411,7 +411,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
           "5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235ac",
           "76a914b5bd079c4d57cc7fc28ecf8213a6b791625b818388ac"}},
         std::nullopt);
-    Check("pk(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+    Check("pk(6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
           "pk("
           "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b"
           "8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)",
@@ -420,7 +420,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235"
             "ac"}},
           std::nullopt);
-    Check("pkh(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+    Check("pkh(6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
           "pkh("
           "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b"
           "8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)",
@@ -429,13 +429,13 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
 
     // Some unconventional single-key constructions
     Check(
-        "sh(pk(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1))",
+        "sh(pk(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe))",
         "sh(pk("
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd))",
         SIGNABLE, {{"a9141857af51a5e516552b3086430fd8ce55f7c1a52487"}},
         OutputType::LEGACY);
     Check(
-        "sh(pkh(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1))",
+        "sh(pkh(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe))",
         "sh(pkh("
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd))",
         SIGNABLE, {{"a9141a31ad23bf49c247dd531a623c2ef57da3c400c587"}},
@@ -543,8 +543,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
         "Key path value '1aa' is not a valid uint32");
 
     // Multisig constructions
-    Check("multi(1,L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1,"
-          "5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+    Check("multi(1,QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe,"
+          "6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
           "multi(1,"
           "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd,"
           "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b"
@@ -555,8 +555,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "40c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abe"
             "a23552ae"}},
           std::nullopt);
-    Check("sortedmulti(1,L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1,"
-          "5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+    Check("sortedmulti(1,QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe,"
+          "6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
           "sortedmulti(1,"
           "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd,"
           "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b"
@@ -567,8 +567,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "40c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abe"
             "a23552ae"}},
           std::nullopt);
-    Check("sortedmulti(1,5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss,"
-          "L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+    Check("sortedmulti(1,6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V,"
+          "QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
           "sortedmulti(1,"
           "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b"
           "8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235,"
@@ -683,8 +683,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
         "P2SH script is too large, 547 bytes is larger than 520 bytes");
     // Invalid threshold
     CheckUnparsable(
-        "multi(a,L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1,"
-        "5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+        "multi(a,QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe,"
+        "6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
         "multi(a,"
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd,"
         "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8d"
@@ -692,8 +692,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
         "Multi threshold 'a' is not valid");
     // Threshold of 0
     CheckUnparsable(
-        "multi(0,L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1,"
-        "5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+        "multi(0,QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe,"
+        "6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
         "multi(0,"
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd,"
         "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8d"
@@ -701,8 +701,8 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
         "Multisig threshold cannot be 0, must be at least 1");
     // Threshold larger than number of keys
     CheckUnparsable(
-        "multi(3,L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1,"
-        "5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)",
+        "multi(3,QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe,"
+        "6KrtvRhkbvGqJ1J46uePDDvWFnrdVauPb9yFkgecqoMVyv8NJ4V)",
         "multi(3,"
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd,"
         "04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8d"
@@ -739,7 +739,7 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
         "KyBXTPy4T7YG4q9tcAM3LkvfRpD1ybHMvcJ2ehaWXaSqeGUxEdkP,"
         "KzJDe9iwJRPtKP2F2AoN6zBgzS7uiuAwhWCfGdNeYJ3PC1HNJ8M8,"
         "L1xbHrxynrqLKkoYc4qtoQPx6uy5qYXR5ZDYVYBSRmCV5piU3JG9,"
-        "L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1))",
+        "QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe))",
         "sh(multi(16,"
         "03669b8afcec803a0d323e9a17f3ea8e68e8abe5a278020a929adbec52421adbd0,"
         "0260b2003c386519fc9eadf2b5cf124dd8eea4c4e68d5e154050a9346ea98ce600,"
@@ -765,20 +765,20 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
 
     // P2SH needs a script, not a key
     CheckUnparsable(
-        "sh(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)",
+        "sh(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)",
         "sh("
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd)",
         "A function is needed within P2SH");
     // Old must be top level
     CheckUnparsable(
         "sh(combo("
-        "L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1))",
+        "QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe))",
         "sh(combo("
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd))",
         "Cannot have combo in non-top level");
     // Cannot embed P2SH inside P2SH
     CheckUnparsable(
-        "sh(sh(pk(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1)))",
+        "sh(sh(pk(QWFEAp2sT88tA4JGh5N12x3A8HQ45AeEhkbJ7p13CPbUunhfAjLe)))",
         "sh(sh(pk("
         "03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd))"
         ")",
diff --git a/src/test/dstencode_tests.cpp b/src/test/dstencode_tests.cpp
index 9603a97203f9ff2881a549f62fa9d81c13b81f56..b00361642b6ae9336e2524f31b81f06f188d5bd8 100644
--- a/src/test/dstencode_tests.cpp
+++ b/src/test/dstencode_tests.cpp
@@ -40,8 +40,8 @@ BOOST_AUTO_TEST_CASE(test_addresses) {
         "ecash:qpm2qsznhks23z7629mms6s4cwef74vcwva87rkuu2";
     std::string cashaddr_script =
         "ecash:ppm2qsznhks23z7629mms6s4cwef74vcwv2zrv3l8h";
-    std::string base58_pubkey = "1BpEi6DfDAUFd7GtittLSdBeYJvcoaVggu";
-    std::string base58_script = "3CWFddi6m4ndiGyKqzYvsFYagqDLPVMTzC";
+    std::string base58_pubkey = "DFxLFMAJWaNYA7TVTUstzPMFRSevAwTSLq";
+    std::string base58_script = "A3FWNUmzq8fXceLoG8DM7PAxPQbNUjae2B";
 
     DummyCashAddrConfig config;
 
diff --git a/src/test/key_tests.cpp b/src/test/key_tests.cpp
index 1567118e82c55a5e0979a44b9a4c6657b59011dc..e6555d007f761b0c9836674874052a10e82f5768 100644
--- a/src/test/key_tests.cpp
+++ b/src/test/key_tests.cpp
@@ -23,21 +23,21 @@
 using util::ToString;
 
 static const std::string strSecret1 =
-    "5HxWvvfubhXpYYpS3tJkw6fq9jE9j18THftkZjHHfmFiWtmAbrj";
+    "6JGrDt8sATpcbAFbFGvjYif4d9hhE8Qitr1ytDvBVRTMKTeM6gK";
 static const std::string strSecret2 =
-    "5KC4ejrDjv152FGwP386VD1i2NYc5KkfSMyv1nGy1VGDxGHqVY3";
+    "6KWPwhKBJgHs4ri6aRk56pzwVo29aT2w3Y79LGurq9Trku6s8Bf";
 static const std::string strSecret1C =
-    "Kwr371tjA9u2rFSMZjTNun2PXXP3WPZu2afRHTcta6KxEUdm1vEw";
+    "QPExFrhiQmPA5mqQA1JAnzrzzZQcYwhhuqML561jJSwJgQae13KH";
 static const std::string strTestSecret1C =
-    "cND2ZvtabDbJ1gucx9GWH6XT9kgTAqfb6cotPt5Q5CyxVDhid2EN";
+    "cfNHiFcNwXRZxiwA4J4HsSiRed5kqgD3RW49Wnmb4AVP3tASBfvX";
 static const std::string strSecret2C =
-    "L3Hq7a8FEQwJkW1M2GNKDW28546Vp5miewcCzSqUD9kCAXrJdS3g";
-static const std::string addr1 = "1QFqqMUD55ZV3PJEJZtaKCsQmjLT6JkjvJ";
-static const std::string addr2 = "1F5y5E5FMc5YzdJtB9hLaUe43GDxEKXENJ";
-static const std::string addr1C = "1NoJrossxPBKfCHuJXT4HadJrXRE9Fxiqs";
-static const std::string addr2C = "1CRj2HyM1CXWzHAXLQtiGLyggNT9WQqsDs";
+    "QUgkGQwEV2RRz2QPcYD76irjY684rduXYCJ7n5EJwWMYcTuqT3DW";
+static const std::string addr1 = "DUPwNcQrNVTmaPUq39t8ry31es4kPtV24v";
+static const std::string addr2 = "DKE4cV1tf1yqXdVUujgu8EoevPxFWYkHdv";
+static const std::string addr1C = "DSwQQ4pXFo5cCCUW37ScqLnujf9XTqAFvn";
+static const std::string addr2C = "DGZpZYuzJcRoXHM84ztGp79HZWBSowYjKM";
 
-static const std::string strAddressBad = "1HV9Lc3sNHZxwj4Zk6fB38tEmBryq2cBiF";
+static const std::string strAddressBad = "DMdEsrzWfhUFUjFAUgejau3qeKbHCy6GGG";
 
 // get r value produced by ECDSA signing algorithm
 // (assumes ECDSA r is encoded in the canonical manner)
diff --git a/src/test/rpc_tests.cpp b/src/test/rpc_tests.cpp
index 2a7443562da5d588df93e7e3fe5eb81b8cda7ddf..64f334ee68b82fdcc9c0029353de638d3c23058d 100644
--- a/src/test/rpc_tests.cpp
+++ b/src/test/rpc_tests.cpp
@@ -284,12 +284,12 @@ BOOST_AUTO_TEST_CASE(rpc_rawsign) {
                           "abd8bcd4f122c6155b1b21027e940bb73ab8732bfdf7f9216ece"
                           "fca5b94d6df834e77e108f68e66f126044c052ae\"}]";
     r = CallRPC(std::string("createrawtransaction ") + prevout + " " +
-                "{\"3HqAe9LtNBjnsfM4CyYaWTnvCaUYT7v4oZ\":11}");
+                "{\"A8aRNzQnSFcgn2iXd7CzkbRHu9raZFHSkT\":11}");
     std::string notsigned = r.get_str();
     std::string privkey1 =
-        "\"KzsXybp9jX64P5ekX1KUxRQ79Jht9uzW7LorgwE65i5rWACL6LQe\"";
+        "\"QSGT8Sd8z8aBcc3o7HAGqeEicLjTCU8JzbVmUZcvp4hCx66fsLPE\"";
     std::string privkey2 =
-        "\"Kyhdf5LuKTRx4ge69ybABsiUAWjVRK4XGxAKk2FQLp2HjGMy87Z4\"";
+        "\"QR6Yov9ta4v5JD38kFRx56Z5dYm4TsCLACrEXeeF5AdeBCNKeB44\"";
     r = CallRPC(std::string("signrawtransactionwithkey ") + notsigned + " [] " +
                 prevout);
     BOOST_CHECK(r.get_obj().find_value("complete").get_bool() == false);
@@ -316,12 +316,12 @@ BOOST_AUTO_TEST_CASE(rpc_rawsign_missing_amount) {
                           "abd8bcd4f122c6155b1b21027e940bb73ab8732bfdf7f9216ece"
                           "fca5b94d6df834e77e108f68e66f126044c052ae\"}]";
     r = CallRPC(std::string("createrawtransaction ") + prevout + " " +
-                "{\"3HqAe9LtNBjnsfM4CyYaWTnvCaUYT7v4oZ\":11}");
+                "{\"A8aRNzQnSFcgn2iXd7CzkbRHu9raZFHSkT\":11}");
     std::string notsigned = r.get_str();
     std::string privkey1 =
-        "\"KzsXybp9jX64P5ekX1KUxRQ79Jht9uzW7LorgwE65i5rWACL6LQe\"";
+        "\"QSGT8Sd8z8aBcc3o7HAGqeEicLjTCU8JzbVmUZcvp4hCx66fsLPE\"";
     std::string privkey2 =
-        "\"Kyhdf5LuKTRx4ge69ybABsiUAWjVRK4XGxAKk2FQLp2HjGMy87Z4\"";
+        "\"QR6Yov9ta4v5JD38kFRx56Z5dYm4TsCLACrEXeeF5AdeBCNKeB44\"";
 
     bool exceptionThrownDueToMissingAmount = false,
          errorWasMissingAmount = false;
diff --git a/src/test/sigcache_tests.cpp b/src/test/sigcache_tests.cpp
index a60ae5eda8789263eb7ad8fc18964ccbf6c70373..43409764197cdb45cc6753aff26934b17c1726fd 100644
--- a/src/test/sigcache_tests.cpp
+++ b/src/test/sigcache_tests.cpp
@@ -20,9 +20,9 @@
 #include <vector>
 
 static const std::string strSecret1 =
-    "5HxWvvfubhXpYYpS3tJkw6fq9jE9j18THftkZjHHfmFiWtmAbrj";
+    "6JGrDt8sATpcbAFbFGvjYif4d9hhE8Qitr1ytDvBVRTMKTeM6gK";
 static const std::string strSecret1C =
-    "Kwr371tjA9u2rFSMZjTNun2PXXP3WPZu2afRHTcta6KxEUdm1vEw";
+    "QPExFrhiQmPA5mqQA1JAnzrzzZQcYwhhuqML561jJSwJgQae13KH";
 
 /**
  * Sigcache is only accessible via CachingTransactionSignatureChecker
diff --git a/src/test/util_tests.cpp b/src/test/util_tests.cpp
index 02c8ab5b96c0edde23cff39bd722cbb91c2b673c..5b01c64719537ff0edd60735c51fe9826a4a5c2e 100644
--- a/src/test/util_tests.cpp
+++ b/src/test/util_tests.cpp
@@ -3102,39 +3102,39 @@ BOOST_AUTO_TEST_CASE(message_verify) {
                       MessageVerificationResult::ERR_INVALID_ADDRESS);
 
     BOOST_CHECK_EQUAL(
-        MessageVerify(*params, "3B5fQsEXEaV8v6U3ejYc8XaKXAkyQj2MjV",
+        MessageVerify(*params, "A1pv9iJRJeN2pTqX4sD2NfChDk91ZwVQqH",
                       "signature should be irrelevant", "message too"),
         MessageVerificationResult::ERR_ADDRESS_NO_KEY);
 
     BOOST_CHECK_EQUAL(MessageVerify(*params,
-                                    "1KqbBpLy5FARmTPD4VZnDDpYjkUvkr82Pm",
+                                    "DPygj5HcNf4iJTZoo5ZLkyz9ctDE7BFjZB",
                                     "invalid signature, not in base64 encoding",
                                     "message should be irrelevant"),
                       MessageVerificationResult::ERR_MALFORMED_SIGNATURE);
 
     BOOST_CHECK_EQUAL(
-        MessageVerify(*params, "1KqbBpLy5FARmTPD4VZnDDpYjkUvkr82Pm",
+        MessageVerify(*params, "DPygj5HcNf4iJTZoo5ZLkyz9ctDE7BFjZB",
                       "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                       "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=",
                       "message should be irrelevant"),
         MessageVerificationResult::ERR_PUBKEY_NOT_RECOVERED);
 
     BOOST_CHECK_EQUAL(
-        MessageVerify(*params, "15CRxFdyRpGZLW9w8HnHvVduizdL5jKNbs",
+        MessageVerify(*params, "D9LXVWacjEAqsWLXrsmrUFoWc8MdPuYuMP",
                       "IPojfrX2dfPnH26UegfbGQQLrdK844DlHq5157/P6h57WyuS/Qsl+h/"
                       "WSVGDF4MUi4rWSswW38oimDYfNNUBUOk=",
                       "I never signed this"),
         MessageVerificationResult::ERR_NOT_SIGNED);
 
     BOOST_CHECK_EQUAL(
-        MessageVerify(*params, "15CRxFdyRpGZLW9w8HnHvVduizdL5jKNbs",
+        MessageVerify(*params, "D9LXVWacjEAqsWLXrsmrUFoWc8MdPuYuMP",
                       "H3CcyS18y471Ya5WTYAke25spvA5eyBMvAuNhxDkuAjsdNGX+"
                       "NoouNlY1KoselGvF65Werf7OdtKAoOZ9WB17Rc=",
                       "Trust no one"),
         MessageVerificationResult::OK);
 
     BOOST_CHECK_EQUAL(
-        MessageVerify(*params, "1NoJrossxPBKfCHuJXT4HadJrXRE9Fxiqs",
+        MessageVerify(*params, "DSwQQ4pXFo5cCCUW37ScqLnujf9XTqAFvn",
                       "HxO6qEsMrNvPA7QTMUbdzF/uXbW78yPG6gFITses9XVvMdBULIFwrfhN"
                       "f196N+rxnK5eSl8eF3aKe4INunzFJXg=",
                       "Trust me"),
diff --git a/src/wallet/test/psbt_wallet_tests.cpp b/src/wallet/test/psbt_wallet_tests.cpp
index 4101903435a00aa89ce094afff6bc37c4ce93b4b..dc5b256849258ada00ca22004e8f5fbcacf94ada 100644
--- a/src/wallet/test/psbt_wallet_tests.cpp
+++ b/src/wallet/test/psbt_wallet_tests.cpp
@@ -74,7 +74,7 @@ BOOST_AUTO_TEST_CASE(psbt_updater_test) {
     // Mainnet and uncompressed form of
     // cUkG8i1RFfWGWy5ziR11zJ5V4U4W3viSFCfyJmZnvQaUsd1xuF3T
     CKey key =
-        DecodeSecret("5KSSJQ7UNfFGwVgpCZDSHm5rVNhMFcFtvWM3zQ8mW4qNDEN7LFd");
+        DecodeSecret("6KkmbMaRwRY4z77yPwqQuP55xoAtkjYAXgUHJtmfKj311vTGFEJ");
     CPubKey master_pub_key = spk_man->DeriveNewSeed(key);
     spk_man->SetHDSeed(master_pub_key);
     spk_man->NewKeyPool();
diff --git a/test/functional/test_framework/test_node.py b/test/functional/test_framework/test_node.py
index 9c4f6a68539407db01274bdd2149ad62c6930c4d..33887f2d686b0234b4fce707eb6d4f6d66120161 100644
--- a/test/functional/test_framework/test_node.py
+++ b/test/functional/test_framework/test_node.py
@@ -253,15 +253,67 @@ class TestNode:
             "cPiRWE8KMjTRxH1MWkPerhfoHFn5iHPWVK5aPqjW8NxmdwenFinJ",
         ),
     ]
+    PRIV_KEYS_TESTNET = [
+        # address , privkey
+        AddressKeyPair(
+            address="nY8xUFpK7LVNNvbmmGaWY8P2wb4MMphFJT",
+            key="cnyWHLn3E8Cy5Mm5ypVU8VHkGqKkm4E5qbhoUo3vtJcVFy4PwZN7",
+        ),
+        AddressKeyPair(
+            address="ngCJhd8WP5JTaugG5DzHJeQH7r5J16Yac3",
+            key="cn88fJ3muTBSMVbhodCHu7NZwngfcHKNExPaos6g6m4q2hoCXmPX",
+        ),
+        AddressKeyPair(
+            address="nbUzAaZrZpgvWK1DPuorqqphY6ZMYACsp9",
+            key="cm1xFxU2Wx8u3d4tXfwP9fKcvdn9A2tjNDgjs8jpRCjQZPrRvhi1",
+        ),
+        AddressKeyPair(
+            address="ndz7noTheExfHkAaGm9fLXazj9RT8cYoJV",
+            key="co4aTtqUwxXWMxXND9uV8xnCe6F5VwrgWD5moRYjAXXcszJ9y3pu",
+        ),
+        AddressKeyPair(
+            address="ngDnaL2W3SuhPy6QpbQcvx8mULMQLL2NPE",
+            key="chNTLE3R5td6UFqjwjQA35jeTWp2o6deVmXGLnSEbVVFu2b1uVnT",
+        ),
+        AddressKeyPair(
+            address="nqXzshPWm3TbJ22v5bQZU1fzLC5CVUwTZX",
+            key="chk1uz7EgkfbF1NnP5xcudsUybCxy388p9wyHy1PZgJ1CrusvJeR",
+        ),
+        AddressKeyPair(
+            address="nng7NB2XJ1ygZ9GQxq4JHPvEHSBpzyqb5E",
+            key="chX5Mf2jWS3TbBQ9WkRb2wCACdfGAa3fQZgEyFFPsnQfiEBy5Nfa",
+        ),
+        AddressKeyPair(
+            address="niT9RnxU2KA97kDwP1r12Atj4Gwmg3K9xK",
+            key="cjh2ZeFHy7eBeQQ2vmA6r29qxjs7CvNNW2TkzAT4DxjoUSv9uRXT",
+        ),
+        AddressKeyPair(
+            address="ndAKYV3Gn7DxCYsgA5mMXwQWAPod12EVeQ",
+            key="cjxo2RcZQ8gF15VtgkKtEpejuv3k9NoLVuxdX7xsZzzuM6Kwmum6",
+        ),
+        AddressKeyPair(
+            address="ndjs9bECAdpgAu3tnTpDjMm18bw4Q5jhgW",
+            key="cfELmuNoe7jD8ChhRfHToxzoKdsftiwtzUMnTpaeKbg74GSnM8tR",
+        ),
+        AddressKeyPair(
+            address="ncvNFSL6XgnQKDL2YeZGWExM1cTn2W1Zdr",
+            key="ckH6TSPpJ4NWJxMM3mpMSsCCaXVc4wzoLwLjnkHPhvkYKE7F5Cjd",
+        ),
+        AddressKeyPair(
+            address="no6r6d9w5ddpmCFgFiJfdDVoWsUvFD6EMk",
+            key="cgsgeYr7i3HhuK2tcuBST3rmn8BPP7vxpCKqWkRh7LUCCc7y3ELb",
+        ),
+    ]
 
     def get_deterministic_priv_key(self):
         """Return a deterministic priv key in base58, that only depends on the node's index"""
-        num_keys = len(self.PRIV_KEYS)
+        priv_keys = self.PRIV_KEYS_TESTNET if self.chain == 'testnet3' else self.PRIV_KEYS
+        num_keys = len(priv_keys)
         assert self.index < num_keys, (
             f"Only {num_keys} keys are defined, please extend TestNode.PRIV_KEYS if "
             "more are needed."
         )
-        return self.PRIV_KEYS[self.index]
+        return priv_keys[self.index]
 
     def _node_msg(self, msg: str) -> str:
         """Return a modified msg that identifies this node by its index as a debugging aid."""
diff --git a/test/util/data/bitcoin-util-test.json b/test/util/data/bitcoin-util-test.json
index d3789ca28899405e3a24596022c88fe64615ae67..f4dc1ecf173960509ddd494c0ca8008c7c4b1bc2 100644
--- a/test/util/data/bitcoin-util-test.json
+++ b/test/util/data/bitcoin-util-test.json
@@ -145,7 +145,7 @@
   { "exec": "./bitcoin-tx",
     "args":
     ["-create",
-     "outaddr=1:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o:garbage"],
+     "outaddr=1:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd:garbage"],
     "return_code": 1,
     "error_txt": "error: TX output missing or too many separators",
     "description": "Malformed outaddr argument (too many separators). Expected to fail."
@@ -172,8 +172,8 @@
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
      "in=bf829c6bcf84579331337659d31f89dfd138f7f7785802d5501c92333145ca7c:18",
      "in=22a6f904655d53ae2ff70e701a0bbd90aa3975c0f40bfc6cc996a9049e31cdfc:1",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
-     "outaddr=4000000:1P8yWvZW8jVihP1bzHeqfE4aoXNX8AVa46"],
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
+     "outaddr=4000000:DTH54BW9S9Q1EPCCiseQCzEBgf6pTRZv3U"],
     "output_cmp": "txcreate1.hex",
     "description": "Creates a new transaction with three inputs and two outputs"
   },
@@ -184,8 +184,8 @@
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
      "in=bf829c6bcf84579331337659d31f89dfd138f7f7785802d5501c92333145ca7c:18",
      "in=22a6f904655d53ae2ff70e701a0bbd90aa3975c0f40bfc6cc996a9049e31cdfc:1",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
-     "outaddr=4000000:1P8yWvZW8jVihP1bzHeqfE4aoXNX8AVa46"],
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
+     "outaddr=4000000:DTH54BW9S9Q1EPCCiseQCzEBgf6pTRZv3U"],
     "output_cmp": "txcreate1.json",
     "description": "Creates a new transaction with three inputs and two outputs (output in json)"
   },
@@ -273,10 +273,10 @@
     "args":
     ["-create", "nversion=1",
      "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-     "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+     "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
      "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
      "sign=ALL",
-     "outaddr=1000:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+     "outaddr=1000:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "output_cmp": "txcreatesignv1.hex",
     "description": "Creates a new v1 transaction with a single input and a single output, and then signs the transaction"
   },
@@ -285,10 +285,10 @@
     ["-json",
      "-create", "nversion=1",
      "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-     "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+     "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
      "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
      "sign=ALL",
-     "outaddr=1000:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+     "outaddr=1000:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "output_cmp": "txcreatesignv1.json",
     "description": "Creates a new v1 transaction with a single input and a single output, and then signs the transaction (output in json)"
   },
@@ -296,10 +296,10 @@
     "args":
     ["-create",
       "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-      "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+      "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
       "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
       "sign=ALL",
-      "outaddr=1000:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+      "outaddr=1000:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "output_cmp": "txcreatesignv2.hex",
     "description": "Creates a new transaction with a single input and a single output, and then signs the transaction"
   },
@@ -307,10 +307,10 @@
     "args":
     ["-create",
       "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-      "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+      "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
       "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485\",\"vout\":\"0foo\",\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
       "sign=ALL",
-      "outaddr=0.001:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+      "outaddr=0.001:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "return_code": 1,
     "error_txt": "error: prevtxs internal object typecheck fail",
     "description": "Tests the check for invalid vout index in prevtxs for sign"
@@ -319,10 +319,10 @@
     "args":
     ["-create",
       "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-      "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+      "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
       "set=prevtxs:[{\"txid\":\"Zd49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59412\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
       "sign=ALL",
-      "outaddr=0.001:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+      "outaddr=0.001:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "return_code": 1,
     "error_txt": "error: txid must be hexadecimal string (not 'Zd49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59412')",
     "description": "Tests the check for invalid txid due to invalid hex"
@@ -331,10 +331,10 @@
     "args":
     ["-create",
       "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-      "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+      "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
       "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc594\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
       "sign=ALL",
-      "outaddr=0.001:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+      "outaddr=0.001:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "return_code": 1,
     "error_txt": "error: txid must be hexadecimal string (not '4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc594')",
     "description": "Tests the check for invalid txid valid hex, but too short"
@@ -343,10 +343,10 @@
     "args":
     ["-create",
       "in=4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc59485:0",
-      "set=privatekeys:[\"5HpHagT65TZzG1PH3CSu63k8DbpvD8s5ip4nEB3kEsreAnchuDf\"]",
+      "set=privatekeys:[\"6J8csdv3eDrnJcpSEb4shfjMh2JTiG9MKzC1Yfge4Y4GyUsjdM6\"]",
       "set=prevtxs:[{\"txid\":\"4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc5948512\",\"vout\":0,\"scriptPubKey\":\"76a91491b24bf9f5288532960ac687abb035127b1d28a588ac\"}]",
       "sign=ALL",
-      "outaddr=0.001:193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"],
+      "outaddr=0.001:DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"],
     "return_code": 1,
     "error_txt": "error: txid must be hexadecimal string (not '4d49a71ec9da436f71ec4ee231d04f292a29cd316f598bb7068feccabdc5948512')",
     "description": "Tests the check for invalid txid valid hex, but too long"
@@ -385,7 +385,7 @@
     "args":
     ["-create",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
      "outdata=4000000:54686973204f505f52455455524e207472616e73616374696f6e206f7574707574207761732063726561746564206279206d6f646966696564206372656174657261777472616e73616374696f6e2e"],
     "output_cmp": "txcreatedata1.hex",
     "description": "Creates a new transaction with one input, one address output and one data output"
@@ -395,7 +395,7 @@
     ["-json",
      "-create", "nversion=1",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
      "outdata=4000000:54686973204f505f52455455524e207472616e73616374696f6e206f7574707574207761732063726561746564206279206d6f646966696564206372656174657261777472616e73616374696f6e2e"],
     "output_cmp": "txcreatedata1.json",
     "description": "Creates a new v1 transaction with one input, one address output and one data output (output in json)"
@@ -404,7 +404,7 @@
     "args":
     ["-create",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
      "outdata=54686973204f505f52455455524e207472616e73616374696f6e206f7574707574207761732063726561746564206279206d6f646966696564206372656174657261777472616e73616374696f6e2e"],
     "output_cmp": "txcreatedata2.hex",
     "description": "Creates a new transaction with one input, one address output and one data (zero value) output"
@@ -414,7 +414,7 @@
     ["-json",
      "-create",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd",
      "outdata=54686973204f505f52455455524e207472616e73616374696f6e206f7574707574207761732063726561746564206279206d6f646966696564206372656174657261777472616e73616374696f6e2e"],
     "output_cmp": "txcreatedata2.json",
     "description": "Creates a new transaction with one input, one address output and one data (zero value) output (output in json)"
@@ -423,7 +423,7 @@
     "args":
     ["-create",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0:4294967293",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"],
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"],
     "output_cmp": "txcreatedata_seq0.hex",
     "description": "Creates a new transaction with one input with sequence number and one address output"
   },
@@ -432,7 +432,7 @@
     ["-json",
      "-create",
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0:4294967293",
-     "outaddr=180000:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"],
+     "outaddr=180000:D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"],
     "output_cmp": "txcreatedata_seq0.json",
     "description": "Creates a new transaction with one input with sequence number and one address output (output in json)"
   },
diff --git a/test/util/data/tt-delin1-out.json b/test/util/data/tt-delin1-out.json
index 6b54272349eb11677db714f26b7d833b655ca4e5..4983ab1211e76c7a14d030a3594e6d8c57260869 100644
--- a/test/util/data/tt-delin1-out.json
+++ b/test/util/data/tt-delin1-out.json
@@ -196,7 +196,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1E7SGgAZFCHDnVZLuRViX3gUmxpMfdvd2o"
+                    "DJFXow7CYcBWKVjwe1VH4or5f6YetLH1hw"
                 ]
             }
         },
@@ -209,7 +209,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1AtWkdmfmYkErU16d3KYykJUbEp9MAj9Sb"
+                    "DF2cHtiK4xeXPUBhMdK7XWU5UNYSg2KFvt"
                 ]
             }
         }
diff --git a/test/util/data/tt-delout1-out.json b/test/util/data/tt-delout1-out.json
index 43b5052a75593670680e4897425358f6274676ea..0e0ed2f8f8e2c2452f1fbd84101024545ee2161e 100644
--- a/test/util/data/tt-delout1-out.json
+++ b/test/util/data/tt-delout1-out.json
@@ -205,7 +205,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1E7SGgAZFCHDnVZLuRViX3gUmxpMfdvd2o"
+                    "DJFXow7CYcBWKVjwe1VH4or5f6YetLH1hw"
                 ]
             }
         }
diff --git a/test/util/data/tt-locktime317000-out.json b/test/util/data/tt-locktime317000-out.json
index 2a340c4e780e7847486219bb0cbc4d7afb393148..bb62afe5e0832128c0890501a8850e213a35c5d8 100644
--- a/test/util/data/tt-locktime317000-out.json
+++ b/test/util/data/tt-locktime317000-out.json
@@ -205,7 +205,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1E7SGgAZFCHDnVZLuRViX3gUmxpMfdvd2o"
+                    "DJFXow7CYcBWKVjwe1VH4or5f6YetLH1hw"
                 ]
             }
         },
@@ -218,7 +218,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1AtWkdmfmYkErU16d3KYykJUbEp9MAj9Sb"
+                    "DF2cHtiK4xeXPUBhMdK7XWU5UNYSg2KFvt"
                 ]
             }
         }
diff --git a/test/util/data/txcreate1.json b/test/util/data/txcreate1.json
index d60743d3b0ca6c6b16dc0d8cb927144c476f69a0..5c0c28e4bebf0e9c8363523f1209cc6b8d700bed 100644
--- a/test/util/data/txcreate1.json
+++ b/test/util/data/txcreate1.json
@@ -43,7 +43,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"
+                    "D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"
                 ]
             }
         },
@@ -56,7 +56,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "1P8yWvZW8jVihP1bzHeqfE4aoXNX8AVa46"
+                    "DTH54BW9S9Q1EPCCiseQCzEBgf6pTRZv3U"
                 ]
             }
         }
diff --git a/test/util/data/txcreatedata1.json b/test/util/data/txcreatedata1.json
index 435da7b80556ccfbbb51bf7a72df8db8f02cdfe3..3273a9411ff1e513eb3035c84bcbc4b7a94a19d6 100644
--- a/test/util/data/txcreatedata1.json
+++ b/test/util/data/txcreatedata1.json
@@ -25,7 +25,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"
+                    "D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"
                 ]
             }
         },
diff --git a/test/util/data/txcreatedata2.json b/test/util/data/txcreatedata2.json
index 03099ef00cc91d4fab5022f011f0c465692cc31e..98c1a55c4d782d8500119b70fd0307296b8b89b8 100644
--- a/test/util/data/txcreatedata2.json
+++ b/test/util/data/txcreatedata2.json
@@ -25,7 +25,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"
+                    "D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"
                 ]
             }
         },
diff --git a/test/util/data/txcreatedata_seq0.json b/test/util/data/txcreatedata_seq0.json
index 951683411fc2befce836949946201ed5d4a7f2fc..b8c565b0d0dadf3f685b9573231ec3d0146b7f8c 100644
--- a/test/util/data/txcreatedata_seq0.json
+++ b/test/util/data/txcreatedata_seq0.json
@@ -25,7 +25,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"
+                    "D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"
                 ]
             }
         }
diff --git a/test/util/data/txcreatedata_seq1.json b/test/util/data/txcreatedata_seq1.json
index ae6de45fa53b8dec9d7434c2df7c220061518414..2c78abbf4ba8e772a7b8fb452dced052dfc0e164 100644
--- a/test/util/data/txcreatedata_seq1.json
+++ b/test/util/data/txcreatedata_seq1.json
@@ -34,7 +34,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "13tuJJDR2RgArmgfv6JScSdreahzgc4T6o"
+                    "D82zqZA4KqaTPmsGegJ1ACoTXiSJ21NcZd"
                 ]
             }
         }
diff --git a/test/util/data/txcreatemultisig1.json b/test/util/data/txcreatemultisig1.json
index 14d00f5dc6b248ae827fb9b54001a3daa9ab39a7..6516594858454ac720029c7f3e494364dfde9e62 100644
--- a/test/util/data/txcreatemultisig1.json
+++ b/test/util/data/txcreatemultisig1.json
@@ -16,9 +16,9 @@
                 "reqSigs": 2,
                 "type": "multisig",
                 "addresses": [
-                    "1FoG2386FG2tAJS9acMuiDsKy67aGg9MKz",
-                    "1FXtz9KU8JNmQDyHdiEm5HDiALuP3zdHvV",
-                    "14LuavcBbXZYJ6Tsz3cAUQj9SuQoL2xCQX"
+                    "DKwMZJ4jYfwAhJckKCMUFz2vrDqscXPMxB",
+                    "DKfzXQG7RiH3wE9tNJEKd3PK3UdgQshoTP",
+                    "D8V18BYptwTpq6eUidbj2AtkL396fFAR4Y"
                 ]
             }
         }
diff --git a/test/util/data/txcreatemultisig2.json b/test/util/data/txcreatemultisig2.json
index 4dda1e63c7a63694f5eea41dd72c453c9565004e..9b0cfdfd8cb7c2ca71f954e887108e21ddf18e84 100644
--- a/test/util/data/txcreatemultisig2.json
+++ b/test/util/data/txcreatemultisig2.json
@@ -16,7 +16,7 @@
                 "reqSigs": 1,
                 "type": "scripthash",
                 "addresses": [
-                    "34HNh57oBCRKkxNyjTuWAJkTbuGh6jg2Ms"
+                    "9u2dRvBhFGJDfKkT9bZvQSNqJUejBBgsLa"
                 ]
             }
         }
diff --git a/test/util/data/txcreatescript2.json b/test/util/data/txcreatescript2.json
index 5ef09d53e58d12a6bd01d8762d4fd77b5eed0563..f8c295c52929af52905c80a11df97b70bf8357e5 100644
--- a/test/util/data/txcreatescript2.json
+++ b/test/util/data/txcreatescript2.json
@@ -16,7 +16,7 @@
                 "reqSigs": 1,
                 "type": "scripthash",
                 "addresses": [
-                    "3C5QarEGh9feKbDJ3QbMf2YNjnMoiPDhNp"
+                    "A2pfKhJAmDYYDxamTYFmuAAkSMjqqvitJd"
                 ]
             }
         }
diff --git a/test/util/data/txcreatesignv1.json b/test/util/data/txcreatesignv1.json
index 4565742a6ef3f2907752070218dc9a31282d489a..7fcc15c2ef3764d7f130da110c5650277737c0ce 100644
--- a/test/util/data/txcreatesignv1.json
+++ b/test/util/data/txcreatesignv1.json
@@ -25,7 +25,7 @@
                 "reqSigs": 1,
                 "type": "pubkeyhash",
                 "addresses": [
-                    "193P6LtvS4nCnkDvM9uXn1gsSRqh4aDAz7"
+                    "DDBUdbqZjUgVKkQX5ju6KmrUKZZzPu2aZc"
                 ]
             }
         }
